admin_panel/src/commonMain/kotlin/net/thechance/mena/admin_panel/data/remote/client/NetworkClient.kt

```diff
@@ -62,15 +61,16 @@ class NetworkClient(
                 bearer {
                     loadTokens {
                         BearerTokens(
-                            accessToken = settings.accessToken,
-                            refreshToken = settings.refreshToken
+                            refreshToken = settings.refreshToken,
+                            accessToken = settings.accessToken
                         )
                     }
                     refreshTokens {
+                        val newAccessToken = refreshToken()
+                        settings.accessToken = newAccessToken
                         BearerTokens(
-                            refreshToken = adminAuthentication.refreshAccessToken(),
-                            accessToken = settings.accessToken
-
+                            accessToken = newAccessToken,
+                            refreshToken = settings.refreshToken
                         )
                     }
                 }
```

Comment:
What about injecting the authenticationService in the NetworkClient and use it in that block like this?

```suggestion
bearer {
                    loadTokens {
                        BearerTokens(
                            accessToken = authorizationService.getAccessToken(),
                            refreshToken = authorizationService.refreshToken()
                        )
                    }
                    refreshTokens {
                        BearerTokens(
                            accessToken = authorizationService.refreshToken(),
                            refreshToken = authorizationService.refreshToken()
                        )
                    }
                }
```

Reply:
Most likely, the same circular dependency will occur if I use this method, which is why I used field injection. Could you take another look at the code and tell me if you have any modifications?